pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -173,6 +173,7 @@ func (p *LaunchTemplateProvider) createLaunchTemplate(ctx context.Context, optio
 		LaunchTemplateName: aws.String(launchTemplateName(options)),
 		LaunchTemplateData: &ec2.RequestLaunchTemplateData{
 			BlockDeviceMappings: p.blockDeviceMappings(options.BlockDeviceMappings),
+			EbsOptimized:        aws.Bool(options.EBSOptimized),
```

Comment:
I believe you get an error if the launch template turns on ebs optimized and the instance type doesn't support it. I think we'd also need to add a scheduling label so that a user can easily filter for instance types that support EBS optimized.

Reply:
@bwagner5 I believe you're right on that. Is there some special handling of such an error case that we should do in the code? Or documentation we should add (which might already be covered in the struct comment pulled from the public AWS docs which indicates  `this isn't available with all instance types`) ?

Reply:
@bwagner5 There is a comment in the linked file `// Well Known to AWS`. Are these well-known labels documented somewhere? My searches are not finding anything. If there's a well-known label for ebs-optimized I'd like to match it, or if not, then I would put the definition into the `// Well known to Karpenter` section instead.

Reply:
After looking into this more, I doubt there are cases where you'd want both instance types that support EBS optimized and those that don't. In fact, it seems like all new instance types `T3+`,  `i3+`, `*4+` all have EBS Optimized on by default.  Is an ebs-optimized attribute actually necessary? Which instance types are you trying to use where you need to manually turn on ebs optimized?

Reply:
Below are all the instance types that do not support EBS Optimized (`unsupported`) or `supported` but not enabled by default. All other instance types have EBS Optimized on by default. These are all legacy instance types, so I'd recommend that we not add ebsOptimized to the NodeTemplate. 

|Instance type | EBS optimization support|
-- | --
  t1.micro | unsupported
   t2.nano | unsupported
   t2.micro | unsupported
   t2.small | unsupported
   t2.medium | unsupported
   t2.large | unsupported
   t2.xlarge | unsupported
   t2.2xlarge | unsupported
  c1.medium | unsupported
   c3.large | unsupported
   c3.8xlarge | unsupported
   cc2.8xlarge | unsupported
   g2.8xlarge | unsupported
   i2.8xlarge | unsupported
   m1.small | unsupported
   m1.medium | unsupported
   m2.xlarge | unsupported
  m3.medium | unsupported
   m3.large | unsupported
   r3.large | unsupported
   r3.8xlarge | unsupported
   c1.xlarge | supported
   c3.xlarge | supported
   c3.2xlarge | supported
   c3.4xlarge | supported
   g2.2xlarge | supported
   i2.xlarge | supported
   i2.2xlarge | supported
   i2.4xlarge | supported
   m1.large | supported
   m1.xlarge | supported
   m2.2xlarge | supported
   m2.4xlarge | supported
   m3.xlarge | supported
   m3.2xlarge | supported
   r3.xlarge | supported
   r3.2xlarge | supported
   r3.4xlarge | supported

Reply:
Out of that list, t2 are still fairly familiar.
But we can inch folks to use t3 instead

Reply:
Yeah t3's are definitely preferable to t2's, in fact, Karpenter filters out t2's by default unless directly specified in the provisioner spec.

Reply:
> Yeah t3's are definitely preferable to t2's, in fact, Karpenter filters out t2's by default unless directly specified in the provisioner spec. 

Good to know.
Is this documented anywhere?

Reply:
It doesn't look like it, we should definitely add that somewhere.   Here are the filters that we apply by default:

https://github.com/aws/karpenter/blob/f4594705f9d20d3d1568fe5f989112dabbc51579/pkg/cloudprovider/aws/cloudprovider.go#L156-L178
https://github.com/aws/karpenter/blob/f4594705f9d20d3d1568fe5f989112dabbc51579/pkg/cloudprovider/aws/instancetypes.go#L153-L197